### PR TITLE
refactor(STADTPULS-625): Use definitions package

### PR DESCRIPTION
**Pull request by vogelino**

### CODE REVIEW

File: src/lib/hooks/useUniqueUsernameValidation/index.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ const checkIfUsernameIsUnique = async (username?: string): Promise<boolean> => {
   const { data: usersWithMatchingUsernames, error } = await supabase
     .from<definitions["user_profiles"]>("user_profiles")
     .select("id")
-    .ilike("name", `%${username}%`);
+    .ilike("name", `%${username.trim()}%`);
```
Comment (dnsos) [inline src/lib/hooks/useUniqueUsernameValidation/index.ts:10]: Why is the `trim` introduced here?

**Review by dnsos — COMMENTED** [review]
Cool, it's nice that the types are now centralized!

There are a few `trim`s introduced. Why was there a need to have them with the new definitions?

File: src/lib/hooks/useUniqueUsernameValidation/index.ts (hunk 1 above)
Comment (vogelino) [reply src/lib/hooks/useUniqueUsernameValidation/index.ts:10]: This makes it defensive against leading and trailing spaces in the usernames. In a perfect world, this should not be necessary but with this additional step, no problem of leading/trailing slashes should occur.

**Review by dnsos — APPROVED** [review]

**Pull request merged**